lib/ab_test.rb

```diff
@@ -20,13 +20,15 @@ def initialize(
     buckets: {},
     should_log: nil,
     default_bucket: :default,
+    report: nil,
```

Comment:
WDYT of a name like `report_config`? Or even a small struct class to help this be more self-documenting? ex `@report = report ? ReportConfig.new(**report.to_h) : nil`

Reply:
I like that! It's a little clunky since there's two different hash shapes to struct-ify (the report config, and individual query configs), but I think it turned out well in 4c10d2d.